src/Shaders/ShadersInclude/pbrLightFunctions.fx

```diff
@@ -7,72 +7,139 @@ struct lightingInfo
     #endif
 };
 
+struct pointLightingInfo
+{
+    vec3 lightOffset;
+    float lightDistanceSquared;
+
+    float attenuation;
+};
+
+struct spotLightingInfo
+{
+    vec3 lightOffset;
+    float lightDistanceSquared;
+    vec3 directionToLightCenterW;
+
+    float attenuation;
+};
+
+float computeDistanceLightFalloff_Standard(vec3 lightOffset, float range)
+{
+    return max(0., 1.0 - length(lightOffset) / range);
+}
+
+float computeDistanceLightFalloff_Physical(float lightDistanceSquared)
+{
+    return 1.0 / ((lightDistanceSquared + 0.001));
+}
+
+float computeDistanceLightFalloff_GLTF(float lightDistanceSquared, float inverseSquaredRange)
+{
+    const float minDistanceSquared = 0.01*0.01;
+    float lightDistanceFalloff = 1.0 / (max(lightDistanceSquared, minDistanceSquared));
+
+    float factor = lightDistanceSquared * inverseSquaredRange;
+    float attenuation = clamp(1.0 - factor * factor, 0., 1.);
+    attenuation *= attenuation;
+
+    // Smooth attenuation of the falloff defined by the range.
+    lightDistanceFalloff *= attenuation;
+    
+    return lightDistanceFalloff;
+}
+
 float computeDistanceLightFalloff(vec3 lightOffset, float lightDistanceSquared, float range, float inverseSquaredRange)
 {   
     #ifdef USEPHYSICALLIGHTFALLOFF
-        float lightDistanceFalloff = 1.0 / ((lightDistanceSquared + 0.001));
+        return computeDistanceLightFalloff_Physical(lightDistanceSquared);

```

Comment:
Not sure to understand how we can get in here as this function seems to be called on lyrics when no falloff define is set